### PR TITLE
JSON API: add inline docs.

### PR DESCRIPTION
**Pull request by jeherve**

### CODE REVIEW

File: class.json-api-endpoints.php [hunk 1]
```diff
@@ -1324,8 +1389,20 @@ function load_theme_functions() {
...
+		/**
+		 * Fires after the theme is initialized.
+		 * @see https://codex.wordpress.org/Plugin_API/Action_Reference/after_setup_theme
+		 *
+		 * @since 3.2.0
+		 */
 		do_action( 'restapi_theme_after_setup_theme' );
 		$this->copy_hooks( 'init', 'restapi_theme_init', WP_CONTENT_DIR . '/themes' );
+		/**
+		 * Fires on init, after WordPress has finished loading but before any headers are sent.
```
Comment (kraftbj) [inline class.json-api-endpoints.php]: This doesn't fire on `init` per se, but mimics the init hook for theme code being accessed via the REST API (if a theme is declaring a CPT or something like that), if I understand this correctly.

Comment (jeherve) [inline class.json-api-endpoints.php]: Oh, you're probably right! How do you think I should phrase this?

File: class.json-api-endpoints.php [hunk 2]
```diff
@@ -1324,8 +1389,20 @@ function load_theme_functions() {
 
 		// since the stuff we care about (CPTS, post formats, are usually on setup or init hooks, we want to load those)
 		$this->copy_hooks( 'after_setup_theme', 'restapi_theme_after_setup_theme', WP_CONTENT_DIR . '/themes' );
+		/**
+		 * Fires after the theme is initialized.
```
Comment (kraftbj) [inline class.json-api-endpoints.php]: Same notes as the below line items apply here.

File: class.json-api-endpoints.php (hunk 1 above)
Comment (kraftbj) [inline class.json-api-endpoints.php]: That's a good question.

Perhaps like
```
* Fires functions hooked onto `init` by the theme for the purpose of the REST API.
*
* The REST API does not load the theme when processing requests. To enable theme-based functionality, the API will load the '/functions.php', '/inc/jetpack.compat.php', '/inc/jetpack.php', '/includes/jetpack.compat.php files of the theme (parent and child) and copy functions hooked onto 'init' within those files.
```
then the same for the after_theme_setup one?

Comment (jeherve) [inline class.json-api-endpoints.php]: I like that, changing this right now, thank you!

**Comment by kraftbj** [comment]
:+1:

**Pull request merged**